### PR TITLE
docs: private-preview docs for avro comments

### PR DESCRIPTION
**Pull request by moulimukherjee**
### Motivation
This is adding documentation for https://github.com/MaterializeInc/materialize/issues/21557

### Tips for reviewer

I added the docs under private preview. Please do suggest if there's a better way to document this. I did not want to over-complicate it with too many examples but wanted to capture the standard path an user might use.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -  documentation for avro comments

cc @umanwizard @morsapaes

### CODE REVIEW

**Comment by moulimukherjee** [comment]
@MaterializeInc/docs Can somebody take a look at this? The 0.73 release notes https://github.com/MaterializeInc/materialize/pull/22458/files#diff-dc24ced769f1f3aa03be91abf803726011125109cca9f86d3dcdb16acf594b36R18 refer to the section which won't have the updates.

File: doc/user/content/sql/create-sink/kafka.md [hunk 1]
```diff
@@ -147,6 +154,88 @@ SQL type                         | Avro type
...
+There are two ways to add comments to the generated avro schema
+* Any [comments](/sql/comment-on) added to the materialized view used in the sink
+  will be propagated the generated schema's "doc" options. Note, the comments for the avro schema
+  will be frozen when the sink is created. Any further comment updates will not be propagated to the schema.
+* Comments can also be specified while creating the sink in the [CSR connection options](#csr-connection-options)
+  using `[KEY|VALUE] DOC ON [TYPE|COLUMN] <identifier>` option. This will override any materialize comment.
+  Specifying `KEY` or `VALUE` is optional and it specifies whether the avro comment is meant for the generated key
+  or the value schema. If it's not specified, then it's applicable for both.
+  The `<identifier>` can be either
+  * a `TYPE` like materialied view, source, tables or custom type used in the sink,
```
Comment (umanwizard) [inline doc/user/content/sql/create-sink/kafka.md:168]:
```suggestion
  * a `TYPE` like materialied view, source, table or custom type used in the sink,
```

**Review by umanwizard — APPROVED** [review]
@morsapaes might want to check this once she's back from vacation, but I think it's good enough to land for now.

**Comment by moulimukherjee** [comment]
@umanwizard Thanks for taking care of this!

**Pull request merged**